src/v/archival/archival_policy.cc

```diff
@@ -319,7 +319,6 @@ static ss::future<> get_file_range(
         // Subsequent call to segment_reader::data_stream will fail in this
         // case. In order to avoid this we need to make another index lookup
         // to find a lower offset which is committed.
-        auto fsize = segment->reader().file_size();
```

Comment:
I wonder - why do we need doing arithmetic with the file size at all? This still seems error-prone (say, it decreased for some reason, truncation possibly or whatever, while we were reading). We have all the file positions we need (`bytes_to_skip` - `stop_at`) and we can calculate content length from them.

Also, I think as we are reading the segment the second time to actually upload the data, we should verify that we have read the same amount there.

Reply:
fixed